mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,358 @@
+        Column 2: hour(numeric values below 24 accourding to bin_width)
+        Column 3: feature values
+
+    Examples
+    --------
+    >>> test = aggregate_data("Distance",20)
+    >>> print(np.mean(test["Distance"]))
+    531.4500177747973
+    """
+    init = pd.DataFrame(columns=["mouse", "strain", "hour", feature])
+    for i in range(3):
+        for j in range(4):
+            if feature == "Distance":
+                tmp = aggregate_movement(strain=i, mouse=j,
+                                         bin_width=bin_width)
+            else:
+                tmp = aggregate_interval(strain=i, mouse=j,
+                                         feature=feature,
+                                         bin_width=bin_width)
+            tmp = pd.DataFrame(list(tmp.values), index=tmp.index)
+            tmp.columns = [feature]
+            tmp["strain"] = i
+            tmp["mouse"] = j
+            tmp["hour"] = tmp.index.hour + tmp.index.minute / 60
+            init = init.append(tmp)
+    return(init)
+
+
+def seasonal_decomposition(strain, mouse, feature, bin_width, period_length):
+    """
+    Apply seasonal decomposition model on the time series
+    of specified strain, mouse, feature and bin_width.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "IS", "M_AS", "M_IS", "W"}
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    res: statsmodel seasonal decomposition object
+        seasonal decomposition result for the mouse.
+        Check the seasonal decomposition plot by res.plot(),
+        seasonl term and trend term by res.seasonal and
+        res.trend separately.
+
+    Examples
+    --------
+    >>> res = seasonal_decomposition(strain=0, mouse=0, feature="W",
+                                     bin_width=30, period_length = 24)
+    """
+    freq = int(period_length * 60 / bin_width)
+    ts = aggregate_interval(strain=strain, mouse=mouse,
+                            feature=feature, bin_width=bin_width)
+    res = sm.tsa.seasonal_decompose(ts.values, freq=freq, model="additive")
+    return(res)
+
+
+def strain_seasonal(strain, mouse, feature, bin_width, period_length):
+    """
+    Use seansonal decomposition model on the time series
+    of specified strain, mouse, feature and bin_width.
+    return the seasonal term and the plot of seasonal term
+    by mouse of a set of mouses in a strain
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: list, set or tuple
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "IS", "M_AS", "M_IS", "W"}
```

Comment:
Same problem as in last function `seasonal_decomposition`, we are supposed to add the seasonal decomposition analysis for travel distance data.